exchanges/auth/README.md

```diff
@@ -0,0 +1,163 @@
+<h2 align="center">@urql/exchange-auth</h2>
+
+<p align="center"><strong>An exchange for managing authentication in <code>urql</code></strong></p>
+
+`@urql/exchange-auth` is an exchange for the [`urql`](https://github.com/FormidableLabs/urql) GraphQL client which helps handle auth headers and token refresh
+
+## Quick Start Guide
+
+First install `@urql/exchange-auth` alongside `urql`:
+
+```sh
+yarn add @urql/exchange-auth
+# or
+npm install --save @urql/exchange-auth
+```
+
+You'll then need to add the `authExchange`, that this package exposes to your `urql` Client
+
+```js
+import { createClient, dedupExchange, cacheExchange, Operation } from 'urql';
+import { executeExchange } from '@urql/exchange-execute';
+
+/**
+ * your auth state can be anything
+ * this is a generic example with a token and refreshToken, but it can
+ * include token expiry for example if you want to check that in willAuthError
+ * without having to decode the token every time
+ **/
+
+type AuthState = {
+  token: String;
+  refreshToken: String;
+} | null;
+
+const client = createClient({
+  url: 'http://localhost:1234/graphql',
+  exchanges: [
+    dedupExchange,
+    cacheExchange,
+    authExchange({
+        addAuthToOperation: ({
+          authState,
+          operation,
+        }: {
```

Comment:
Typically we don't use types in examples just to keep it smaller, since people will figure this part out intuitively